### DISCUSSION TITLE
Airflow is not able to work with Google dependecies

### DISCUSSION THREAD

**Discussion by pavel93yan** [apache/airflow#34485] [General]
### Apache Airflow version

2.7.1

### What happened

I'm using Airflow that is deployed using Docker I installed  the latest version of apache-airflow-providers-google also tried(10.7.0) but faced with the issue Airflow is not able to use Google hook for Gsheet and other stuff related to Google provider.
It complains on 'google' folder, tells that it's not a package. I also tried to add this folder to PYTHONPATH, it didn't help.

Error:
`  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/google/suite/hooks/sheets.py", line 23, in <module>
    from googleapiclient.discovery import build
  File "/home/airflow/.local/lib/python3.8/site-packages/googleapiclient/discovery.py", line 42, in <module>
    import google.api_core.client_options
ModuleNotFoundError: No module named 'google.api_core'; 'google' is not a package`

Could you please help me with it?

### What you think should happen instead

Google dependencies should be imported correctly

### How to reproduce

```
FROM apache/airflow:2.7.1
COPY airflow_requirements.txt constraints-3.8.txt /
RUN pip install --no-cache-dir "apache-airflow[google,amazon]==${AIRFLOW_VERSION}"  \
 -r /airflow_requirements.txt
ENV PIP_USER=false
RUN python -m venv dbt_venv && source dbt_venv/bin/activate && \
    pip install --no-cache-dir "dbt-postgres==1.6.1" && \
    pip install --no-cache-dir "elementary-data==0.10.0" && \
    pip install --no-cache-dir "elementary-data[postgres]==0.10.0" && \
    deactivate
ENV PIP_USER=true
ENV PYTHONPATH "${PYTHONPATH}:${AIRFLOW_HOME}/utils"
```
Use this in instead of airflow image in official [docker compose](https://airflow.apache.org/docs/apache-airflow/2.7.1/docker-compose.yaml) file.

### Operating System

docker

### Versions of Apache Airflow Providers

astronomer-cosmos==1.1.1
apache-airflow-providers-postgres==5.6.0
apache-airflow-providers-amazon==8.6.0
pandas==2.0.3
pyarrow==11.0.0
pyairtable==2.1.0
apache-airflow-providers-google==10.7.0

### Deployment

Docker-Compose

### Deployment details

```
FROM apache/airflow:2.7.1
COPY airflow_requirements.txt constraints-3.8.txt /
RUN pip install --no-cache-dir "apache-airflow[google,amazon]==${AIRFLOW_VERSION}"  \
 -r /airflow_requirements.txt
ENV PIP_USER=false
RUN python -m venv dbt_venv && source dbt_venv/bin/activate && \
    pip install --no-cache-dir "dbt-postgres==1.6.1" && \
    pip install --no-cache-dir "elementary-data==0.10.0" && \
    pip install --no-cache-dir "elementary-data[postgres]==0.10.0" && \
    deactivate
ENV PIP_USER=true
ENV PYTHONPATH "${PYTHONPATH}:${AIRFLOW_HOME}/utils"
```
Use this in instead of airflow image in official [docker compose](https://airflow.apache.org/docs/apache-airflow/2.7.1/docker-compose.yaml) file.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by tallestJake** [comment]
Hi @pavel93yan did you find a solution to this?
Having same issue with python 3.11 and airflow 2.8.1
